Tree_From_Preorder_and_Inorder_Traversal_in_Python.py

```diff
@@ -0,0 +1,56 @@
+#constructing a node class for tree
+class Node:
+    def __init__(self):
+        self.data=None
+        self.left=None
+        self.right=None
+#constructing a binary tree class
+class BinaryTree:
+    def __init__(self,preorder,ion) :
+        self.root = constructPre(preorder, 0, len(preorder) - 1, ino, 0, len(ino) - 1)
+#function for constructing the tree from given inorder and preorder traversal
+def constructPre(pre,plo,phi,ino,ilo,ihi):
+    root=Node()
+    if (plo > phi or ilo > ihi):
+        return None
+    root.data = pre[plo]
+    si=-1
+    for i in range(ilo,ihi+1):
+        if (ino[i] == pre[plo]):
+            si = i
+            break
+    nel = si - ilo
+    #constructing left subtree with help of recursion
+    root.left = constructPre(pre, plo + 1, plo + nel, ino, ilo, si - 1)
+    root.right = constructPre(pre, plo + nel + 1, phi, ino, si + 1, ihi)
+    return root
+#function for displaying the tree
```

Comment:
Add new line above this and do this for function as well.

Reply:
Sorry not understood what you mean to say @jainaman224 . Please explain again . I am new to open source that's why getting problem.

Reply:
You should add new lines above every function.
So it is easier to understand the code.